### DISCUSSION TITLE
BD_TOPO : "Cannot find a SRID value for the layer commune.shp."

### DISCUSSION THREAD

**Discussion by atoulou** [orbisgis/geoclimate#931] [Q&A] [answered]
Hello,

After using Geoclimate with OSM input data, I'm trying to use it with BD TOPO (for my PhD thesis). I managed to do it with the example given (sample 12174). I'd now like to use the tool for Lyon.

I downloaded the BD TOPO files (version 3.3 here https://geoservices.ign.fr/bdtopo). I checked that there were all the necessary inputs (Aerodrome - Building - Cemetery - Municipality - Surface construction - Transport equipment - Aerodrome runway - Reservoir - Hydrographic surface - Sports ground - Road trunk - Railway trunk - Vegetation zone - Zone of activity or interest) and that for each layer there were the formats '.dbf', '.prj', '.shx' and '.shp'. I then modified the configuration file as follows: 

```json
{
    "description": "Processing BD Topo v3 data",
    "input": {
        "folder":"C:\\Users\\atoulou\\Desktop\\Geoclimate\\BDTOPO_3.3_69",
        "locations":["42059"]
				},
    "output": {
        "folder": "C:\\Users\\atoulou\\Desktop\\Geoclimate\\Resultats_BD_TOPO"
    },
    "parameters": {
        "rsu_indicators": {
            "indicatorUse": [
                "LCZ",
                "TEB",
                "URBAN_TYPOLOGY"
            ],
            "svfSimplified": true
        },
        "grid_indicators": {
            "x_size": 100,
	    "y_size": 100,
	    "rowCol": false,
	    "output" : "geojson",
	    "indicators" :["BUILDING_FRACTION", "BUILDING_HEIGHT", "BUILDING_POP",
                               "BUILDING_TYPE_FRACTION", "WATER_FRACTION", "VEGETATION_FRACTION",
                               "ROAD_FRACTION", "IMPERVIOUS_FRACTION", "FREE_EXTERNAL_FACADE_DENSITY",
                               "BUILDING_HEIGHT_WEIGHTED", "BUILDING_SURFACE_DENSITY",
                               "SEA_LAND_FRACTION", "ASPECT_RATIO", "SVF",
                               "HEIGHT_OF_ROUGHNESS_ELEMENTS", "TERRAIN_ROUGHNESS_CLASS",
                               "UTRF_AREA_FRACTION", "UTRF_FLOOR_AREA_FRACTION",
                               "LCZ_PRIMARY"]
         }
    }
}
```

On my command prompt I wrote : 
PS C:\Users\atoulou\Desktop\Geoclimate> java -jar geoclimate-1.0.0.jar -f my_first_config_file_BDTOPO3.json -w BDTOPO_V3 

"42059" is one of the "communes" for which I have data. 
With this configuration file, I got the message "Cannot find a SRID value for the layer commune.shp".
I don't understand why I got this message or how to solve it ... could you please help me?

Second question: on the website, it says "If you want all the communes in your data to be executed, remove "id_zones":["12174"]." yet when I tried the above configuration file without the line "locations":["42059"] I got this error message "ERROR GeoClimate -- Cannot find any locations parameter". So how do I get the tool to work for the whole area (and not just one "commune")?

Thanks in advance for your help :)

**Comment by j3r3m1** [comment]
Concerning your first issue, you probably need to add an information about the srid of your data since the BDTopo srid is not recognized easily by H2GIS (the "spatial database" system used in GeoClimate): [here you can find more information](https://github.com/orbisgis/geoclimate/wiki/Frequently-Asked-Questions#how-to-force-the-srid-with-the-bdtopo-workflow-)

Thus your config file will result in:
```json
{
    "description": "Processing BD Topo v3 data",
    "input": {
        "folder":"C:\\Users\\atoulou\\Desktop\\Geoclimate\\BDTOPO_3.3_69",
        "locations":["42059"],
        "srid":2154
				},
    "output": {
        "folder": "C:\\Users\\atoulou\\Desktop\\Geoclimate\\Resultats_BD_TOPO"
    },
    "parameters": {
        "rsu_indicators": {
            "indicatorUse": [
                "LCZ",
                "TEB",
                "URBAN_TYPOLOGY"
            ],
            "svfSimplified": true
        },
        "grid_indicators": {
            "x_size": 100,
	    "y_size": 100,
	    "rowCol": false,
	    "output" : "geojson",
	    "indicators" :["BUILDING_FRACTION", "BUILDING_HEIGHT", "BUILDING_POP",
                               "BUILDING_TYPE_FRACTION", "WATER_FRACTION", "VEGETATION_FRACTION",
                               "ROAD_FRACTION", "IMPERVIOUS_FRACTION", "FREE_EXTERNAL_FACADE_DENSITY",
                               "BUILDING_HEIGHT_WEIGHTED", "BUILDING_SURFACE_DENSITY",
                               "SEA_LAND_FRACTION", "ASPECT_RATIO", "SVF",
                               "HEIGHT_OF_ROUGHNESS_ELEMENTS", "TERRAIN_ROUGHNESS_CLASS",
                               "UTRF_AREA_FRACTION", "UTRF_FLOOR_AREA_FRACTION",
                               "LCZ_PRIMARY"]
         }
    }
}
```

**Reply by atoulou** [reply]
Thank you for your reply. I added the line "srid":2154 to my configuration file. It worked with some "communes" but not with all ... For example, it worked with the commune "69200", but with others, such as the commune "69123" it didn't work, I got the following message:
 
Exception lors de l'appel de la fonction définie par l'utilisateur: "intersection(GEOMETRYCOLLECTION (POLYGON ((842137.7 
6515153.1, 842166.3999999997 6515153.1, 842165.6 6515152.3, 842167.1 6515150.7, 842163.9 6515148, 842165.2 6515146.6, 842162.5 6515144.2, 842164.8 6515141.6, 842166.9 6515139.3, 842168 6515138.1, 842159.4 6515130.3, 842158.3 6515131.5, 842156.2 6515133.9, 842156 6515133.8, 842151.8 6515138.5, 842150.9 6515137.6, 842147.1 6515134.3, 842137.7 6515125.857407407, 842137.7 6515153.1)), POINT (842137.7 6515123.1)), POLYGON ((842037.7 6515053.1, 842137.7 6515053.1, 842137.7 6515153.1, 842037.7 6515153.1, 842037.7 6515053.1))): Overlay input is mixed-dimension"; SQL statement: DROP TABLE IF EXISTS upper_table_join [90105-224]

How can I resolve this issue ? 

Thanks again in advance,

**Reply by j3r3m1** [reply]
This one is probably a bug, you cannot solve it but we can.

**Reply by atoulou** [reply]
Ok thanks for your reply, should I tell you all the communes I need and for which I got the same error message? 
And do you have any idea how long it will take to resolve this bug?

Thank you again,

**Reply by j3r3m1** [reply]
Ideally, you can tell us more (if the same error is logged and which insee codes) about this bug in this issue #933

**Comment by j3r3m1** [comment]
Concerning your second issue, good that you found this problem in the wiki. It is a matter of update, I have changed that since this option seems not to be possible anymore. However, you can still run a list of communes. So copying and copying the list of all your insee codes after the "locations" key word should make the job.

**Reply by atoulou** [reply]
Ok thank you, I'll try when I've solved the first issue.

**Comment by j3r3m1** [comment]
@atoulou can we consider this discussion solved ?

**Comment by atoulou** [comment, marked as answer]
Yes! Thank you again for your help
